ubirch/crypto.go

```diff
@@ -189,20 +195,71 @@ func (c *CryptoContext) SetKey(name string, id uuid.UUID, privKeyBytes []byte) e
 // Get a certificate signing request.
 func (c *CryptoContext) GetCSR(name string) ([]byte, error) { return nil, nil }
 
+// Get the decoded public key for the given name.
+func (c *CryptoContext) getDecodedPublicKey(name string) (*ecdsa.PublicKey, error) {
+	id, err := c.GetUUID(name)
+	if err != nil {
+		return nil, err
+	}
+
+	pubKey, err := c.Keystore.GetKey(pubKeyEntryTitle(id))
+	if err != nil {
+		return nil, err
+	}
+
+	// decode the key
+	return decodePublicKey(pubKey)
+}
+
 // Get the public key bytes for the given name.
 func (c *CryptoContext) GetPublicKey(name string) ([]byte, error) {
+	decodedPubKey, err := c.getDecodedPublicKey(name)
+	if err != nil {
+		return nil, fmt.Errorf("decoding public key from keystore failed: %s", err)
+	}
+
+	pubKeyBytes := make([]byte, 0, 0)
+
+	paddedX := make([]byte, 32)
+	paddedY := make([]byte, 32)
+	copy(paddedX[32-len(decodedPubKey.X.Bytes()):], decodedPubKey.X.Bytes())
+	copy(paddedY[32-len(decodedPubKey.Y.Bytes()):], decodedPubKey.Y.Bytes())
+	pubKeyBytes = append(pubKeyBytes, paddedX...)
+	pubKeyBytes = append(pubKeyBytes, paddedY...)
+
+	return pubKeyBytes, nil
```

Comment:
Maybe also check len(pubkeybytes)==expectedPubkeySize before returning to be sure? (To catch errors introduced if code above is changed later.)

Reply:
at this point len(pubkeybytes) will allways be 64 since we just filled in the 2 32byte padded byte arrays

Reply:
Yes, if everything works and there is no bug in the code above/no one fiddles with the code later. Just an extra measure, maybe its a bit paranoid.